src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -418,41 +450,63 @@ class IpexCommunicationService extends AgentService {
     ).map((key) => previousExnGrantMsg.exn.e.acdc.e?.[key]?.s);
     allSchemaSaids.push(schemaSaid);
 
-    const op = await Agent.agent.multiSigs.multisigAdmit(
+    const { op } = await this.multisigService.multisigAdmit(
       holder.signifyName,
       previousExnGrantMsg.exn.d as string,
       allSchemaSaids,
       multisigExn
     );
 
     const schema = await this.props.signifyClient.schemas().get(schemaSaid);
-    await this.saveAcdcMetadataRecord(
-      previousExnGrantMsg.exn.e.acdc.d,
-      previousExnGrantMsg.exn.e.acdc.a.dt,
-      schema.title,
-      connectionId,
-      schemaSaid
-    );
+    const credentialPending =
+      await this.credentialStorage.getCredentialMetadata(
+        previousExnGrantMsg.exn.e.acdc.d
+      );
 
-    this.props.eventService.emit<AcdcStateChangedEvent>({
-      type: AcdcEventTypes.AcdcStateChanged,
-      payload: {
-        credentialId,
-        status: CredentialStatus.PENDING,
-      },
-    });
+    if (!credentialPending) {
+      await this.saveAcdcMetadataRecord(
+        previousExnGrantMsg.exn.e.acdc.d,
+        previousExnGrantMsg.exn.e.acdc.a.dt,
+        schema.title,
+        connectionId,
+        schemaSaid
+      );
 
-    const pendingOperation = await this.operationPendingStorage.save({
-      id: op.name,
-      recordType: OperationPendingRecordType.ExchangeReceiveCredential,
+      this.props.eventService.emit<AcdcStateChangedEvent>({
+        type: AcdcEventTypes.AcdcStateChanged,
+        payload: {
+          credentialId,
+          status: CredentialStatus.PENDING,
+        },
+      });
+
+      const pendingOperation = await this.operationPendingStorage.save({
```

Comment:
I think this should be outside this if statement. So that if we accept multiple admits, we have an operation for each.

I know right now KERIA will use the same ID and override but that will change that, so I'm worried we will forget here to go back and track all of them